### DISCUSSION TITLE
Milvus search_iterator expr max size ?

### DISCUSSION THREAD

**Discussion by mistborn17** [milvus-io/milvus#36904] [Q&A and General discussion]
What is the max size of the expr supported by the Milvus search_iterator API ?

**Comment by liliu-z** [comment]
Did you mean the expr's string length? @mistborn17

**Reply by mistborn17** [reply]
Yes. The max string length of expr.

**Reply by liliu-z** [reply]
> Yes. The max string length of expr.

We don't have any specific change but the gRPC has size limit, which can affect this

**Comment by yhmo** [comment]
The size of each RPC request is limited to 64MB by default on the server side, configured in the milvus.yaml. So, the max size of expr must be less than this value, else you will get a RPC error.
```
proxy:
  grpc:
    serverMaxRecvSize: 67108864 # The maximum size of each RPC request that the proxy can receive, unit: byte
```

**Comment by xiaofan-luan** [comment]
We don't recommend to use huge expr with 1MB. or more.

@xiaocai2333 is working a expr template feature which could potentially increase the limit but yes 64MB is the upper limit
